### PR TITLE
Add coding guidelines for Streams API

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for reviews, @guozhangwang @dguy @bbejeck

File: coding-guide.html [hunk 1]
```diff
@@ -102,6 +102,47 @@
...
+                            This holds for all class members, local variables, loop variables, and method parameters.</li>
+			<li>Write modular and thus testable code. Refactor if necessary!</li>
+			<li>Avoid large PRs (recommended is not more the 500 lines per PR).
+                            Many JIRAs requires larger code changes; thus, split the work in multiple PRs and create according sub-task on the JIRA to track the work.</li>
+			<li>All public APIs must have JavaDocs.</li>
+			<li>Verify if JavaDocs are still up to date or if they need to be updated.</li>
+			<li>JavaDocs: Write grammatically correct sentences and use punctuation marks correctly.</li>
+			<li>Use proper markup (e.g., <code>{@code null}</code>).</li>
+			<li>Update the documentation on the Kafka webpage (i.e., within folder <code>docs/</codes>.
+                            Doc changes are not additional work (i.e. no follow up PRs) but part of the actuall PR (can also be a sub-tasks).</li>
```
Comment (kamalcph) [inline coding-guide.html]: @mjsax 

nit: actuall -> actual

Comment (Sharon2017) [reply coding-guide.html]: Two more nits:
> All contributors should follow those the get a high quality and uniform code base.

Nit: follow those the get -> follow those to get

>  thus make the live of all contributors easier

Nit: the live -> the life

Comment (mjsax) [reply coding-guide.html]: Thanks @Kamal15 @Sharon2017 !

**Comment by guozhangwang** [comment]
LGTM. Merged to `asf-site`.

**Pull request closed without merging**